### PR TITLE
add multicodecs list to libp2p spec

### PR DESCRIPTION
**Pull request by diasdavid**
Writing these down in the spec so that we agree and don't loose them again.

### CODE REVIEW

File: libp2p/7-properties.md [hunk 1]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/
+| yamux | /yamux/1.0.0 |
+| multiplex | /multiplex/6.7.0 |
+| spdy | /spdy/3.1.0 |
+| dht | /ipfs/dht/1.0.0 | 
+| bitswap | /ipfs/bitswap/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: @whyrusleeping found:

```
exchange/bitswap/network/interface.go
11:var ProtocolBitswap protocol.ID = "/ipfs/bitswap"
```

Any reason to not have a version?

File: libp2p/7-properties.md [hunk 2]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/
+| yamux | /yamux/1.0.0 |
+| multiplex | /multiplex/6.7.0 |
+| spdy | /spdy/3.1.0 |
+| dht | /ipfs/dht/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: Same here:

```
routing/dht/dht.go
32:var ProtocolDHT protocol.ID = "/ipfs/dht"
```

File: libp2p/7-properties.md [hunk 3]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/
+| yamux | /yamux/1.0.0 |
+| multiplex | /multiplex/6.7.0 |
+| spdy | /spdy/3.1.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: ✔️
```
p2p/net/swarm/swarm.go
49:     msstpt.AddTransport("/spdy/3.1.0", spdy.Transport)
```

File: libp2p/7-properties.md [hunk 4]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+
+Here, we present a table with the multicodecs defined for each IPFS protocol that has a wire componenent. This list may change overtime and currently exists as a guide for implementation
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/
+| yamux | /yamux/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: ✔️
```
p2p/net/swarm/swarm.go
48:     msstpt.AddTransport("/yamux/1.0.0", ymxtpt)
```

File: libp2p/7-properties.md [hunk 5]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/1.0.0 |
+| yamux | /yamux/1.0.0 |
+| multiplex | /multiplex/6.7.0 |
+| spdy | /spdy/3.1.0 |
+| dht | /ipfs/dht/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: Same here:

```
routing/dht/dht.go
32:var ProtocolDHT protocol.ID = "/ipfs/dht"

File: libp2p/7-properties.md [hunk 6]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/1.0.0 |
+| yamux | /yamux/1.0.0 |
+| multiplex | /multiplex/6.7.0 |
+| spdy | /spdy/3.1.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: ✔️
```
p2p/net/swarm/swarm.go
49:     msstpt.AddTransport("/spdy/3.1.0", spdy.Transport)
```

File: libp2p/7-properties.md [hunk 7]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+
+Here, we present a table with the multicodecs defined for each IPFS protocol that has a wire componenent. This list may change overtime and currently exists as a guide for implementation
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/1.0.0 |
+| yamux | /yamux/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: ✔️
```
p2p/net/swarm/swarm.go
48:     msstpt.AddTransport("/yamux/1.0.0", ymxtpt)
```

File: libp2p/7-properties.md [hunk 8]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+### 7.5.7 Protocol Multicodecs
+
+Here, we present a table with the multicodecs defined for each IPFS protocol that has a wire componenent. This list may change overtime and currently exists as a guide for implementation
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
+| multistream | /ipfs/multistream/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: Do we want multistream to be:

```
/multistream/1.0.0
```

or 

```
/ipfs/multistream/1.0.0
```

File: libp2p/7-properties.md [hunk 9]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+
+### 7.5.7 Protocol Multicodecs
+
+Here, we present a table with the multicodecs defined for each IPFS protocol that has a wire componenent. This list may change overtime and currently exists as a guide for implementation
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
+| identify | /ipfs/identify/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: This has changed to `/ipfs/id/1.0.0`, which is also what I remember from the discussion.  Could you confirm? 

```
p2p/protocol/identify/id.go
25:const ID = "/ipfs/id/1.0.0"
```

File: libp2p/7-properties.md [hunk 10]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
 The wire protocol is -- of course -- wrapped with encryption. We use cyphersuites similar to TLS. This is explained further in the [network spec](./#encryption).
+
+### 7.5.7 Protocol Multicodecs
+
+Here, we present a table with the multicodecs defined for each IPFS protocol that has a wire componenent. This list may change overtime and currently exists as a guide for implementation
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
+| plain    | /plaintext/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: ✔️

p2p/net/conn/listen.go
24:     NoEncryptionTag = "/plaintext/1.0.0"

File: libp2p/7-properties.md [hunk 11]
```diff
@@ -190,3 +190,19 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
 
 The wire protocol is -- of course -- wrapped with encryption. We use cyphersuites similar to TLS. This is explained further in the [network spec](./#encryption).
+
+### 7.5.7 Protocol Multicodecs
+
+Here, we present a table with the multicodecs defined for each IPFS protocol that has a wire componenent. This list may change overtime and currently exists as a guide for implementation
+
+| protocol | multicodec |
+| :------: | :--------: |
+| secio    | /ipfs/secio/1.0.0 |
```
Comment (diasdavid) [inline libp2p/7-properties.md]: Found:

```
p2p/net/conn/listen.go
23:     SecioTag        = "/secio/1.0.0"
```

Which makes sense to not be scoped by `/ipfs`. Just confirm please.

**Comment by diasdavid** [comment]
@whyrusleeping could you confirm this list and see if I also missed a protocol?

File: libp2p/7-properties.md (hunk 8 above)
Comment (whyrusleeping) [inline libp2p/7-properties.md]: I think multistream should just be `/muiltistream/1.0.0`. Its not ipfs specific.

On that same note, identify and secio arent necessarily ipfs specific either.

File: libp2p/7-properties.md (hunk 11 above)
Comment (whyrusleeping) [inline libp2p/7-properties.md]: Yeap, I beleive that is correct. from this doc: https://gist.github.com/whyrusleeping/c1b53200dc2072564478f1b9e3671371#gistcomment-1743999

File: libp2p/7-properties.md (hunk 1 above)
Comment (whyrusleeping) [inline libp2p/7-properties.md]: Hrm... that appears to be a typo. I remember making a commit where I added versions to this, diag and the dht. Where did that go?

File: libp2p/7-properties.md (hunk 8 above)
Comment (diasdavid) [inline libp2p/7-properties.md]: +1  to `/multistream/1.0.0`

File: libp2p/7-properties.md (hunk 11 above)
Comment (diasdavid) [inline libp2p/7-properties.md]: I looked for this everywhere!!! ahaha :D

Comment (whyrusleeping) [inline libp2p/7-properties.md]: confirmed.

File: libp2p/7-properties.md (hunk 8 above)
Comment (whyrusleeping) [inline libp2p/7-properties.md]: lets go with just /multistream/1.0.0

**Comment by diasdavid** [comment]
@whyrusleeping I've updated these according to the gist. Is it good to update the ones in go to also use semver?

File: libp2p/7-properties.md [hunk 12]
```diff
@@ -190,3 +190,22 @@ For now, we use [protobuf](https://github.com/google/protobuf) for all protocol
...
+plaintext | /plaintext/1.0.0 
+spdy | /spdy/3.1.0
+yamux | /yamux/1.0.0
+multiplex | /multiplex/6.7.0
+identify | /ipfs/id/1.0.0
+ping | /ipfs/ping/1.0.0
+relay | /ipfs/relay/line/1.0.0
+diagnostics | /ipfs/diag/net/1.0.0
+Kademlia DHT | /ipfs/kad/1.0.0
+bitswap | /ipfs/bitswap/1.0.0
```
Comment (diasdavid) [inline libp2p/7-properties.md]: @whyrusleeping can you ping me here when you get to merge the double handlers on go-ipfs to support the version with semver? thank you :) 

//cc @dignifiedquire

**Pull request merged**